Broken link fixes to 2.2 documents

fstrr opened the pull request:
- fixed links to Accessible Auth (Minimum) in Accessible Auth (Enhanced)
- Updated understanding Understanding:
  - fix all in-page links
  - replace references to WCAG 2.1 with WCAG 2.2
- fix links to Accessible Auth criteria in Redundant Entry 
- fix broken links in techniques → about. (side note: that list hasn't been updated in over three years)
- fix links to 2.2 criteria

patrickhlauke commented:
With apologies for @fstrr I think I redid the same thing in https://github.com/w3c/wcag/pull/3738

patrickhlauke commented:
@fstrr this now has massive merge conflicts. wondering if this has been superseded by other site-wide changes we made?

fstrr commented:
@patrickhlauke woah. yeah. I'll take a look and see. hopefully it has. thanks.

fstrr commented:
@patrickhlauke I merged in the current main branch and dealt with the merge conflicts. I also fixed some typos, some out-of-date links, added a couple of links, and mostly succeeded in not tidying up whitespace.

bruce-usab reviewed (approved):
Thank you for another labor of love!

mbgower commented on techniques/failures/F42.html:
```diff
@@ -116,13 +116,13 @@ <h3>Scripting a <code class="language-html">div</code> element</h3>
          <ul>
             <li>
-                  <a href="https://www.w3.org/TR/wai-aria/">Accessible Rich Internet Applications (WAI-ARIA) Version 1.0</a>
+                  <a href="https://www.w3.org/TR/wai-aria/">Accessible Rich Internet Applications (WAI-ARIA) Version</a>
```
since we just want it to point at the most recent version, I have removed the word "Version"
```suggestion
                  <a href="https://www.w3.org/TR/wai-aria/">Accessible Rich Internet Applications (WAI-ARIA)</a>
```
